### DISCUSSION TITLE
How can I aggregate calculated field?

### DISCUSSION THREAD

**Discussion by ricemountainer** [prisma/prisma#21237] [Q&A] [answered]
### Question

Hi experts! 
Please let me ask a question. I want to get the result like this:

| id | num |
|--- |------ |
| 1  | 76     |
| 2  | 0       |

...from the input table like this:

| id | num1 | num2 |
|--- |------- |------- |
| 1  | 10      | 11       |
| 1  | 22      | 33       |
| 2  | 1        | -1       |
| 2 | 2         | -2       |

The `num` field in the result is calculated by `sum(num1 + num2)` in the input table. I mean, I want to query like `select id, sum(num1+num2) from input group by id` without writing raw query. Regarding this, I couldn't find out the doc or code sample. Can I write the code in Prisma?

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: Ubuntu "20.04.6 LTS (Focal Fossa)"
- Database: PostgreSQL 13.12 (Ubuntu 13.12-1.pgdg20.04+1) on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0, 64-bit
- Node.js version: v18.17.1

- Run `prisma -v` to see your Prisma version and paste it

```
prisma                  : 4.15.0
@prisma/client          : 4.12.0
Current platform        : debian-openssl-1.1.x
Query Engine (Node-API) : libquery-engine 8fbc245156db7124f997f4cecdd8d1219e360944 (at node_modules/@prisma/engines/libquery_engine-debian-openssl-1.1.x.so.node)
Migration Engine        : migration-engine-cli 8fbc245156db7124f997f4cecdd8d1219e360944 (at node_modules/@prisma/engines/migration-engine-debian-openssl-1.1.x)
Format Wasm             : @prisma/prisma-fmt-wasm 4.15.0-28.8fbc245156db7124f997f4cecdd8d1219e360944
Default Engines Hash    : 8fbc245156db7124f997f4cecdd8d1219e360944
Studio                  : 0.484.0
Done in 15.07s.
```

**Comment by ludralph** [comment, marked as answer]
Hi @ricemountainer 👋 

Thank you for raising this question.

Prisma currently does not support operations like `SUM(VALUE1 + VALUE2)` directly in the `groupBy` function. However, you can use the groupBy function to group by ID and then calculate the sum of VALUE1 and VALUE2 separately. Here's an example:
```js
const result = await prisma.input.groupBy({
  by: ['id'],
  _sum: {
    num1: true,
    num2: true,
  
  },
});
```

This will return an object where each ID is associated with the sum of num1 and num2. You can then add these sums together in your application code.
```
[
  { _sum: { num1: 3, num2: -3 }, id: 2 },
  { _sum: { num1: 32, num2: 44 }, id: 1 }
]
```
```js
const res = result.map(item => ({
  id: item.id,
  num: item._sum.num1 + item._sum.num2,
}));
```
The result then becomes:
| id | num |
|--- |------ |
| 2  | 0     |
| 1  | 76       |

You can create a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) to see this supported natively in Prisma.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by ricemountainer** [reply]
I understand. I think it is a bit difficult to implement the feature but that definitely makes me happy so much. I'll request later!

Actually what I really want to do is to calculate the delta between two `timestamp` columns then aggregate the result. Probably it has a bit different from your code sample above, but I think a part of your solution can be used for that. I'll try later. Thank you so much!

**Reply by ludralph** [reply]
You're welcome 🙌
